src/psyclone/psyGen.py

```diff
@@ -1708,6 +1744,30 @@ def update(self):
                 text += " COLLAPSE({0})".format(self._collapse)
         self._add_region(start_text=text)
 
+    def begin_string(self):
```

Comment:
Could the gen_code() and update() methods use this string instead of duplicating the logic that generates the string?
I see the DirectiveGen expect separate parameters but maybe some unwrapped list like: `*self.begin_string().split()` would work? If it turns out not to be that easy that won't mind the duplication.

Can you also consider this in the other Directives that this PR created begin_ and end_ strings for?

Reply:
Oh yes, I was trying to avoid touching `update()` since it will be removed soon but actually it was easy to tidy both it and `gen_code`.

Reply:
I couldn't do this in `ACCParallelDirective` because, as you say, DirectiveGen expects separate parameters and I can't pass it a list without modifying `DirectiveGen` itself. Similarly for `ACCKernelsDirective` and `ACCDataDirective`.

Reply:
I was going to resolve this issue because as you mention this will be removed soon and is not a big deal, but since there is still another comment in the review, what I meant here was:
```parent.add(DirectiveGen(parent, *self.begin_string().split()))```
and below:
```parent.add(DirectiveGen(parent, *self.end_string().split(), ""))```

These should works without modifying the DirectiveGen

Reply:
Oh yes, I live and learn! That's nice.